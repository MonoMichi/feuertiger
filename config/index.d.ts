export interface Config {
    projectName: string;
    gitlab: {
        projectId: string;
        token: string;
        user: string;
        email: string;
        repositoryUrl: string;
        branch: string;
        branchSlug: string;
        commit: string;
    };
    npmRegistry: string;
    gcp: {
        project: string;
        region: string;
        zone: string;
        accessToken: string;
    };
    digitaloceanToken: string;
    firebaseAppConfig: {
        projectId: string;
        apiKey: string;
        authDomain: string;
    };
    firebaseAdminConfig: {
        projectId: string;
        privateKey: string;
        clientEmail: string;
    };
    postgresUri: string;
    graphqlUri: string;
    webClientUri: string;
}

declare const config: Config;

export default config;
