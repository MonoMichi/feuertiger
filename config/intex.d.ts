export default interface Config {
    projectName: string;
    gitlab: {
        projectId: string;
        token: string;
        user: string;
    };
    gcp: {
        project: string;
        region: string;
        zone: string;
        accessToken: string;
    };
    digitaloceanToken: string;
    gitBranchSlug: string;
    gitCommitHashShort: string;
    firebaseAppConfig: {
        projectId: string;
        apiKey: string;
        authDomain: string;
    };
    firebaseAdminConfig: {
        projectId: string;
        privateKey: string;
        clientEmail: string;
    };
    postgresUri: string;
    graphqlUri: string;
    webClientUri: string;
}
