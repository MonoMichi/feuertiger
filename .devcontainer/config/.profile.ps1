$theme = 'https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/atomicBit.omp.json'
$file = '~/.config/powershell/atomicBit.json'

if (-not (Test-Path $file)) {
	curl -sS $theme > $file
}

oh-my-posh init pwsh --config $file | Invoke-Expression

$cmdlets = Resolve-Path -Path "$($PSScriptRoot)/../../tools/cmdlets/feuertiger.psd1"
Import-Module -Name $cmdlets