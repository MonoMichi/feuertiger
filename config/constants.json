{
    "projectName": "feuertiger",
    "gitlab": {
        "projectId": "18726494"
    },
    "gcp": {
        "project": "feuertiger-77a8a",
        "region": "europe-west1",
        "zone": "eur3"
    }
}
