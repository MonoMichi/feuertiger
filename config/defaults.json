{
    "postgresUri": "postgresql://feuertiger:feuertiger@localhost:5432/feuertiger",
    "graphqlUri": "http://localhost:4000/",
    "webClientUri": "http://localhost:3000/"
}
