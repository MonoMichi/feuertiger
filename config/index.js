const { execSync } = require('child_process');
const constants = require('./constants.json');

/*
 * Get all existing environment variables
 */
const FIREBASE_CONFIG = JSON.parse(process.env.FIREBASE_CONFIG);
const GOOGLE_CREDENTIALS = JSON.parse(process.env.GOOGLE_CREDENTIALS);

const {
    POSTGRES_URI,
    GRAPHQL_URI,
    WEB_CLIENT_URI,
    DIGITALOCEAN_TOKEN,
    GIT_TOKEN
} = process.env;

/*
 * Compute configs
 */
exports = {
    projectName: constants.projectName,
    gitlab: {
        ...constants.gitlab,
        token: GIT_TOKEN,
        user: GIT_USER
    },
    gcp: constants.gcp,
    digitaloceanToken: DIGITALOCEAN_TOKEN,
    gitBranchSlug: execSync('git branch --show-current')
        .toString('utf-8')
        .trim()
        .toLowerCase()
        .replace(/[^a-z^0-9]/g, '-'),
    gitCommitHashShort: execSync('git rev-parse --short HEAD')
        .toString('utf-8')
        .trim(),
    firebaseAppConfig: {
        projectId: FIREBASE_CONFIG.project_id,
        apiKey: FIREBASE_CONFIG.apiKey,
        authDomain: FIREBASE_CONFIG.authDomain
    },
    firebaseAdminConfig: {
        projectId: GOOGLE_CREDENTIALS.project_id,
        privateKey: GOOGLE_CREDENTIALS.private_key,
        clientEmail: GOOGLE_CREDENTIALS.client_email
    },
    postgresUri:
        POSTGRES_URI ||
        'postgresql://feuertiger:feuertiger@localhost:5432/feuertiger',
    graphqlUri: GRAPHQL_URI || 'http://localhost:4000/',
    webClientUri: WEB_CLIENT_URI || 'http://localhost:3000'
};

/*
 * Re-export all deafault environment variables
 */
execSync(`export WEB_CLIENT_URI=${exports.webClientUri}`);
execSync(`export GRAPHQL_URI=${exports.graphqlUri}`);
execSync(`export POSTGRES_URI=${exports.postgresUri}`);

execSync(`export CYPRESS_BASE_URL=${exports.webClientUri}`);

execSync(`export GIT_BRANCH_SLUG=${exports.gitBranchSlug}`);
execSync(`export GIT_COMMIT_HASH_SHORT=${exports.gitCommitHashShort}`);
