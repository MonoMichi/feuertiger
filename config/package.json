{
    "name": "@feuertiger/config",
    "version": "1.0.9-dev.fc5afdd.0",
    "author": "Michael Thanei",
    "license": "MIT",
    "description": "config",
    "main": "./dist/index.js",
    "scripts": {
        "build": "tsc",
        "dev": "tsc --watch --pretty --noEmit | awk '{gsub(/\\033c/,\"\") system(\"\")}1'"
    },
    "repository": {
        "type": "git",
        "url": "https://gitlab.com/feuertiger/feuertiger",
        "directory": "config"
    },
    "publishConfig": {
        "@feuertiger:registry": "https://gitlab.com/api/v4/projects/18726494/packages/npm/"
    },
    "devDependencies": {
        "@types/node": "14.14.22"
    },
    "gitHead": "05e4076488965ce7af8aec66f20dae4fb035d3b0",
    "sideEffects": false
}
