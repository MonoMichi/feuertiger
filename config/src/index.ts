/* eslint-disable camelcase */
import constants from './constants.json';
import defaults from './defaults.json';
import {
    tryGetGitBranchSlug,
    tryGetGitCommitHashShort,
    tryGetGitUserEmail,
    tryGetGitUserName
} from './utils';

const {
    POSTGRES_USER,
    POSTGRES_PASSWORD,
    POSTGRES_URI,
    GRAPHQL_URI,
    WEB_CLIENT_URI,
    DIGITALOCEAN_TOKEN,
    GIT_TOKEN,
    GIT_USER,
    GIT_EMAIL,
    FIREBASE_CONFIG: FIREBASE_CONFIG_TEXT,
    GOOGLE_CREDENTIALS: GOOGLE_CREDENTIALS_TEXT
} = process.env;

const FIREBASE_CONFIG = JSON.parse(FIREBASE_CONFIG_TEXT ?? '{}');

const GOOGLE_CREDENTIALS = JSON.parse(GOOGLE_CREDENTIALS_TEXT ?? '{}');

export const { projectName, gcp, dockerRegistry } = constants;

export const gitlab = {
    ...constants.gitlab,
    token: GIT_TOKEN,
    user: GIT_USER ?? tryGetGitUserName(),
    email: GIT_EMAIL ?? tryGetGitUserEmail(),
    repositoryUrl: '',
    branch: '',
    branchSlug: tryGetGitBranchSlug(),
    commit: tryGetGitCommitHashShort()
};

export const npmRegistry = `//gitlab.com/api/v4/projects/${constants.gitlab.projectId}/packages/npm/`;

export const dockerRegistryRepository = `${constants.dockerRegistry}/${constants.projectName}/${constants.projectName}`;

export const digitaloceanToken = DIGITALOCEAN_TOKEN;

export const firebaseAppConfig = {
    projectId: FIREBASE_CONFIG?.projectId,
    apiKey: FIREBASE_CONFIG?.apiKey,
    authDomain: FIREBASE_CONFIG?.authDomain
};

export const firebaseAdminConfig = {
    projectId: GOOGLE_CREDENTIALS?.project_id,
    privateKey: GOOGLE_CREDENTIALS?.private_key,
    clientEmail: GOOGLE_CREDENTIALS?.client_email
};

export const postgresUser = POSTGRES_USER ?? defaults.postgresUser;

export const postgresPassword = POSTGRES_PASSWORD ?? defaults.postgresPassword;

export const postgresUri = POSTGRES_URI ?? defaults.postgresUri;

export const graphqlUri = GRAPHQL_URI ?? defaults.graphqlUri;

export const webClientUri = WEB_CLIENT_URI ?? defaults.webClientUri;

export const env = {
    GIT_USER: gitlab.user,
    GIT_EMAIL: gitlab.email,
    POSTGRES_URI: postgresUri,
    GRAPHQL_URI: graphqlUri,
    WEB_CLIENT_URI: webClientUri
};
