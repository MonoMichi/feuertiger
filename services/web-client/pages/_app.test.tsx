import { test, describe, expect, beforeAll } from '@jest/globals';
import App, { AppProps } from './_app';
import renderer from 'react-test-renderer'; // jest.setup.js
import { setConfig } from 'next/config';
import config from '../next.config';

const appPropsMock: AppProps = {
    auth: {
        signInWithEmailAndPassword: () => {},
        signOut: () => {}
    },
    error: undefined,
    apollo: {},
    isLoading: false,
    isSignedIn: true,
    pageProps: {}
};

describe('Test nextjs app', () => {
    beforeAll(() => setConfig(config));
    test('should create _app', () => {
        const component = renderer.create(
            <App {...appPropsMock}>
                <div>test</div>
            </App>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
