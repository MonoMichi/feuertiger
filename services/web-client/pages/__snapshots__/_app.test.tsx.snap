// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test nextjs app should create _app 1`] = `
<div
  className="container__RootDiv-jkdao3-2 iIvLLv"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary appBar__StyledAppBar-sc-1d678t4-0 gFvmkR mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular appBar__StyledToolbar-sc-1d678t4-1 feaBTq MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root appBar__StyledIconButton-sc-1d678t4-2 fxQcjk MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <h6
        className="MuiTypography-root appBar__StyledTypography-sc-1d678t4-3 iHhMhi MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
      >
        Dashboard
      </h6>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
  <div
    className="MuiDrawer-root MuiDrawer-docked sideBar__StyledDrawer-la5i2z-0 cTMWBV"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <div
        className="sideBar__StyledDiv-la5i2z-1 fWvBvl"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          data-testid="sidebar-close-button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <hr
        className="MuiDivider-root"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div>
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Dashboard
              </span>
            </div>
            <span
              className="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.67 13.13C18.04 14.06 19 15.32 19 17v3h4v-3c0-2.18-3.57-3.47-6.33-3.87z"
                  fillRule="evenodd"
                />
                <circle
                  cx="9"
                  cy="8"
                  fillRule="evenodd"
                  r="4"
                />
                <path
                  d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4c-.47 0-.91.1-1.33.24C14.5 5.27 15 6.58 15 8s-.5 2.73-1.33 3.76c.42.14.86.24 1.33.24zM9 13c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
                  fillRule="evenodd"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Mitglieder
              </span>
            </div>
            <span
              className="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Ãœbungen
              </span>
            </div>
            <span
              className="MuiTouchRipple-root"
            />
          </div>
        </div>
      </ul>
      <hr
        className="MuiDivider-root"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div />
      </ul>
    </div>
  </div>
  <main
    className="container__StyledMain-jkdao3-0 ewHDMg"
  >
    <div
      className="MuiContainer-root container__StyledContainer-jkdao3-1 eCBlrR MuiContainer-maxWidthLg"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": 40,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style={
          Object {
            "height": 190,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": 40,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style={
          Object {
            "height": 190,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": 40,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style={
          Object {
            "height": 190,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": 40,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style={
          Object {
            "height": 190,
            "width": undefined,
          }
        }
      />
    </div>
  </main>
</div>
`;
